src/elf/reloc.rs

```diff
@@ -183,12 +189,9 @@ macro_rules! elf_rela_std_impl { ($size:ident, $isize:ty) => {
 
             pub fn from_fd(fd: &mut File, offset: usize, size: usize) -> Result<Vec<Rela>> {
                 let count = size / SIZEOF_RELA;
-                let mut bytes = vec![0u8; size];
+                let mut relocs = vec![Rela::default(); count];
```

Comment:
Actually, Rust prohibits any such function from being considered safe.
It's [undefined](https://doc.rust-lang.org/reference.html#behavior-considered-undefined) for any code to read uninitialized memory, even though reading an arbitrary bit pattern from `Plain` type is well-defined. So code using uninitialized vectors would always be unsafe "de jure".